src/n64sys.c

```diff
@@ -352,6 +357,29 @@ void wait_ms( unsigned long wait_ms )
     wait_ticks(TICKS_FROM_MS(wait_ms));
 }
 
+/**
+ * @brief Force a complete halt of all processors
+ *
+ * @note It should occur whenever a reset has been triggered 
+ * and its past its RESET_TIME_LENGTH grace time period.
+ * This function will shut down the RSP and the CPU, blank the VI.
+ * Eventually the RDP will flush and complete its work as well.
+ * The system will recover after a reset or power cycle.
+ * 
+ */
+__attribute__((noreturn)) void die(){
+    // Halt the RSP
+    *SP_STATUS = SP_WSTATUS_SET_HALT;
+    // Flush the RDP
+    *DP_STATUS = DP_WSTATUS_SET_FLUSH | DP_WSTATUS_SET_FREEZE;
+    // Shut the video off
+    *VI_CTRL = *VI_CTRL & (~VI_CTRL_TYPE);
+    // Can't have any interrupts here
+    disable_interrupts();
+    // Spin infinitely
+    while(true){};
```

Comment:
Please call `abort()` here, so that the intent is clear. This way we also have a single abort point that might be useful for debugging (eg: gdb can autoset a breakpoint in abort() to catch when the program aborts for any reason).